### PR TITLE
Reduced CPU load when using RequestFuture

### PR DESCRIPTION
**Pull request by Vaerys-Dawn**
### Prerequisites
* [x] I have read and understood the [contribution
guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
* [Proof of testing](http://i.imgur.com/1eohpnE.gif)

**Issues Fixed:** Reduced CPU usage when using Requestbuffer.request().get();

### Changes Proposed in this Pull Request
* Addes 50ms sleep to while loop within both .get() overloads found
within the RequestBuffer.request() subclass.

### CODE REVIEW

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
 	 * Queues a request.
 	 *
 	 * @param request The request to be carried out.
-	 * @param <T> The type of the object returned by the request.
+	 * @param <T>     The type of the object returned by the request.
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Why is this changed?

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 2]
```diff
@@ -136,7 +136,8 @@ public static int killAllRequests() {
 		 *
 		 * @param requestFuture The future managing this request.
 		 */
-		default void onRetry(RequestFuture<T> requestFuture) {}
+		default void onRetry(RequestFuture<T> requestFuture) {
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Don't change/reformat the entire file.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 3]
```diff
@@ -183,7 +185,7 @@ public long getDelay(TimeUnit unit) {
 			if (isDone() || isCancelled())
 				return 0;
 
-			return unit.convert(callable.timeForNextRequest-System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+			return unit.convert(callable.timeForNextRequest - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Don't change/reformat the entire file.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 4]
```diff
@@ -197,7 +199,7 @@ public String getBucket() {
 
 		@Override
 		public int compareTo(Delayed o) {
-			return (int) (getDelay(TimeUnit.MILLISECONDS)-o.getDelay(TimeUnit.MILLISECONDS));
+			return (int) (getDelay(TimeUnit.MILLISECONDS) - o.getDelay(TimeUnit.MILLISECONDS));
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Don't change/reformat the entire file.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 5]
```diff
@@ -255,7 +261,7 @@ private void run() {
 			backing.run();
 		}
 
-		private static class RequestCallable <T> implements Callable<T> {
+		private static class RequestCallable<T> implements Callable<T> {
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Don't change/reformat the entire file.

File: src/main/java/sx/blah/discord/util/RequestBuffer.java [hunk 6]
```diff
@@ -284,7 +290,7 @@ public T call() {
 					}
 				} catch (RateLimitException e) {
 					firstAttempt = false;
-					timeForNextRequest = System.currentTimeMillis()+e.getRetryDelay();
+					timeForNextRequest = System.currentTimeMillis() + e.getRetryDelay();
```
Comment (chrislo27) [inline src/main/java/sx/blah/discord/util/RequestBuffer.java]: Don't change/reformat the entire file.

**Comment by austinv11** [comment]
Closing this in favor of f2e5004
